src/python/WMCore/Services/Requests.py

```diff
@@ -163,6 +168,29 @@ def makeRequest_pycurl(self, uri=None, params={}, verb='GET',
         #And now overwrite any headers that have been passed into the call:
         headers.update(incoming_headers)
         url = self['host'] + uri
+
+        # NOTE: VK 20151026
+        # this block is added intentionally to immitate makeRequest_httplib
+        # behavior. THIS IS BAD design though since it modifies HTTP requests
+        # and instead of passing dictionary parameters it encodes them.
+        # But we need to leave it as is until WMCore APIs will be adjusted.
+        if verb != 'GET' and params:
+            if isinstance(encoder, type(self.get)) or hasattr(encoder, '__call__'):
+                encoded_data = encoder(params)
+            elif encoder == False:
+                # Don't encode the data more than we have to
+                #  we don't want to URL encode the data blindly,
+                #  that breaks POSTing attachments... ConfigCache_t
+                #encoded_data = urllib.urlencode(data)
+                #  -- Andrew Melo 25/7/09
+                encoded_data = params
+            else:
+                # Either the encoder is set to True or it's junk, so use
+                # self.encode
+                encoded_data = self.encode(params)
+            if isinstance(encoded_data, basestring):
+                headers["Content-length"] = str(len(encoded_data))
```

Comment:
Content-length need to be recalculated in param is dict. since param is passed below not encoded_data in line 194